content.go

```diff
@@ -191,3 +197,69 @@ func FetchBytes(ctx context.Context, target ReadOnlyTarget, reference string, op
 
 	return desc, bytes, nil
 }
+
+// PushBytes describes the contentBytes using the given mediaType and pushes it.
+// If mediaType is not specified, "application/octet-stream" will be used.
+func PushBytes(ctx context.Context, pusher content.Pusher, mediaType string, contentBytes []byte) (ocispec.Descriptor, error) {
+	desc := content.NewDescriptorFromBytes(mediaType, contentBytes)
+	r := bytes.NewReader(contentBytes)
+	if err := pusher.Push(ctx, desc, r); err != nil {
+		return ocispec.Descriptor{}, err
+	}
+
+	return desc, nil
+}
+
+// defaultTagConcurrency is the default concurrency of tagging.
+const defaultTagConcurrency = 5 // This value is consistent with dockerd.
+
+// TagBytes describes the contentBytes using the given mediaType, pushes it,
+// and tag it with the given references.
+// If mediaType is not specified, "application/octet-stream" will be used.
+func TagBytes(ctx context.Context, target Target, mediaType string, contentBytes []byte, references ...string) (ocispec.Descriptor, error) {
```

Comment:
I've noticed `defaultTagConcurrency`. Should we have a `TagBytesOptions`?

Reply:
Refactored.